File: weyl-application/src/main/java/io/quartic/weyl/CatalogueClientConfiguration.java
package io.quartic.weyl;

public class CatalogueClientConfiguration {
    private String hostName;
    private Integer port;
    private Boolean useSsl;

    public void setHostName(String hostName) {
        this.hostName = hostName;
    }

    public void setPort(Integer port) {
        this.port = port;
    }

    public void setUseSsl(Boolean useSsl) {
        this.useSsl = useSsl;
    }

    public String getRestUrl() {
        return String.format("%s://%s:%d/api/", useSsl ? "https" : "http", hostName, port);
    }

    public String getWatchUrl() {
        return String.format("%s://%s:%d/api/datasets/watch", useSsl ? "wss" : "ws", hostName, port);
    }
}

File: catalogue-utils/src/main/kotlin/io/quartic/catalogue/CatalogueClientConfiguration.kt
package io.quartic.catalogue

class CatalogueClientConfiguration {
    var hostname: String? = null
    var port: Int? = null
    var useSsl: Boolean? = null

    val restUrl: String
        get() = "${if (useSsl!!) "https" else "http"}://$hostname:$port/api/"

    val watchUrl: String
        get() = "${if (useSsl!!) "wss" else "ws"}://$hostname:$port/api/datasets/watch"
}
